howto-addons.md

```diff
@@ -0,0 +1,118 @@
+This section explores more advanced use cases with addons. For an introduction, see [the addon tutorial](TODO: link to the tutorial page)
```

Comment:
You suggested to have a separate section for addons, and I think that makes a lot of sense. We didn't have it yet since the current addon page had all how-tos on one page, and I hadn't cleaned it up yet. 

I suggest to have a wrapper page for addons that should contain a short explanation on what they are and what they should be used for, plus links to the tutorial and reference. And then each of the sections below can be a separate how-to - except for the Best practices, those don't really work as how-to. For now, I think we could just put them on the wrapper page. We can split them out later if we end up with an explanation page for addons.